### PR TITLE
[DADZ-290] Adds HD intermediate rotation task

### PR DESCRIPTION
**Pull request by 0xpApaSmURf**

### CODE REVIEW

File: tasks/keyInfraSetters.ts [hunk 1]
```diff
@@ -35,6 +36,40 @@ task("rotate:IntermediateKey")
     }
   });
 
+task("hd:rotate:IntermediateKey")
+  .addParam("verifiercontract", "Address of the Access Token Verifier Contract")
+  .addParam("newintermediate", "Address of the new intermediate key")
+  .setAction(async function (taskArguments: TaskArguments, { ethers }) {
+    const newIntermediateKey = taskArguments.newintermediate;
+    if (!newIntermediateKey || newIntermediateKey === ethers.constants.AddressZero) {
+      throw new Error("Invalid new intermediate key");
```
Comment (sudoFerraz) [inline tasks/keyInfraSetters.ts:45]: Is there a specific length to this key? If yes would be great if we also add that on this check

File: tasks/keyInfraSetters.ts [hunk 2]
```diff
@@ -35,6 +36,40 @@ task("rotate:IntermediateKey")
...
+    }
+
+    const ledger = new LedgerSigner(ethers.provider);
+
+    const accessTokenVerifier = <AccessTokenVerifier>(
+      await ethers.getContractAt("AccessTokenVerifier", taskArguments.verifiercontract)
+    );
+
+    let tx: ContractTransaction | null = null;
```
Comment (sudoFerraz) [inline tasks/keyInfraSetters.ts:54]: Interesting notation, never saw it before, could you clarify it for my curiosity? :)

**Review by sudoFerraz — APPROVED** [review]

**Comment by sudoFerraz** [comment]
There is something weird on this repo, this test that failed on the CI does not have anything to do with what you changed on this PR

File: tasks/keyInfraSetters.ts (hunk 2 above)
Comment (0xpApaSmURf) [reply tasks/keyInfraSetters.ts:54]: Maybe stupid but just very explicit declaration here.

Using the `|` notation as type union to declare the potential types of `tx` as both `ContractTransaction` and `null` and then initialising it to `null`.

File: tasks/keyInfraSetters.ts (hunk 1 above)
Comment (0xpApaSmURf) [reply tasks/keyInfraSetters.ts]: Fixed in https://github.com/violetprotocol/ethereum-access-token/pull/33/commits/4e2b573cdcef62759550bdd9f9131d3eaa8e71d9

**Review by sudoFerraz — APPROVED** [review]

**Review by sudoFerraz — APPROVED** [review]

**Pull request merged**